src/client/components/dashboard/CreateNewModal.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { FC, useState } from 'react'
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  Button,
+  Input,
+  Text,
+} from '@chakra-ui/react'
+
+import { Checker } from '../../../types/checker'
+import { ApiClient } from '../../api'
+
+export type CreateNewModalProps = {
+  isOpen: boolean
+  onClose: () => void
+  onSuccess: () => void
+  checker?: Checker
+}
+
+export const CreateNewModal: FC<CreateNewModalProps> = ({
+  isOpen,
+  onClose,
+  onSuccess,
+  checker,
+}) => {
+  const initial = {
+    id: '',
+    title: '',
+    description: '',
+    fields: [],
+    constants: [],
+    operations: [],
+    displays: [],
+  }
+
+  const [newChecker, setChecker] = useState<Checker>(
+    checker
+      ? {
+          ...checker,
+          id: initial.id,
+          title: initial.title,
+          description: initial.description,
+        }
+      : initial
+  )
+
+  const [idInvalid, setIdInvalid] = useState(false)
+  const [titleInvalid, setTitleInvalid] = useState(false)
+  const [errorText, setErrorText] = useState('')
+
+  const setId = (event: React.FocusEvent<HTMLInputElement>) => {
+    const id = event.target.value
+    const invalid = !/^[a-z0-9-]+[a-z0-9]+$/.test(id)
+    setIdInvalid(invalid)
+    setErrorText(
+      invalid ? 'Id must be lowercase letters and numbers, separated by -' : ''
+    )
+    if (!invalid) {
+      setChecker({ ...newChecker, id })
+    }
+  }
+
+  const setTitle = (event: React.FocusEvent<HTMLInputElement>) => {
+    const title = event.target.value
+    const invalid = title === ''
+    setTitleInvalid(invalid)
+    if (!invalid) {
+      setChecker({ ...newChecker, title })
+    }
+  }
+
+  const setDescription = (event: React.FocusEvent<HTMLInputElement>) => {
+    const description = event.target.value
+    if (!description) {
+      const checkerWithoutDescription = { ...newChecker }
+      delete checkerWithoutDescription.description
+      setChecker(checkerWithoutDescription)
+    } else {
+      setChecker({ ...newChecker, description })
+    }
+  }
+
+  const resetAndClose = () => {
+    setChecker(initial)
+    setErrorText('')
+    onClose()
+  }
+
+  const onSubmit = async () => {
+    try {
+      await ApiClient.post('/c', newChecker)
+      onSuccess()
+      resetAndClose()
+    } catch (error) {
+      setErrorText(error.response.data.message)
+    }
+  }
+
+  return (
+    <Modal isOpen={isOpen} onClose={resetAndClose}>
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>
+          {checker ? `Copy ${checker.id}` : 'Create New Checker'}
+        </ModalHeader>
+        <ModalCloseButton />
+        <ModalBody>
+          <Input
```

Comment:
Minor point, maybe want to consider using [react-hook-form](https://react-hook-form.com/) instead to save on  boilerplate? It's already added for the main checker form. 

```jsx
const { register, handleSubmit, errors, formState } = useForm()
const { isValid, isValidating } = formState

// Submit returns { id: ..., title: ... }
const onSubmit = (data) => console.log(data);

// ...

<form onSubmit={handleSubmit(onSubmit)}>
  <FormControl isInvalid={errors.title}>
     <Input name="title" ref={register({ required: true })} />
     <FormErrorMessage>{errors.title && errors.title.message}</FormErrorMessage>
  </FormControl>

  <Button disabled={!isValid} type="submit">Create</Button>
</form>
```

Reply:
I'll look into this